lnwallet/channel.go

```diff
@@ -2974,6 +2998,20 @@ func (lc *LightningChannel) evaluateHTLCView(view *HtlcView,
 			)
 			if rmvHeight == 0 {
 				switch {
+				// If this a noop add, then when we settle the
+				// HTLC, we actually credit the sender with the
+				// amount again, thus making it a noop.
+				case entry.EntryType == Settle &&
+					addEntry.EntryType == NoopAdd:
+
+					delta := int64(entry.Amount)
+					balanceDeltas.ModifyForParty(
```

Comment:
Also important to mention that parties could disagree here, with receiver potentially trying out malicious things (like going to chain, like @ZZiigguurraatt mentioned), and try to "grab" both the custom data and the btc amount

That's a good reason why I believe we should not convert the entry type to `NoopAdd` for HTLCs that carry a btc amount above a certain threshold, or let the user explicitly mark it somehow as `noop`

Reply:
Not sure I follow, but if they disagree, a force close occurs. Which doesn't benefit either party, as they need to pay for chain fees, and now the receiver will likely end up with nothing as it'll cost more to spend the output than it's actually worth.

Reply:
I mean that we should not allow significant btc amounts being carried by HTLCs of type `NoopAdd`

That breaks the incentive alignment and someone could actually want to misbehave and force-close/pay fees